packages/terra-graphs/src/components/PairedResult/PairedResultGraph.jsx

```diff
@@ -0,0 +1,59 @@
+/* eslint import/no-unresolved: off */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Carbon from '@cerner/carbon-graphs/lib/js/carbon';
+import '../Graph.module.scss';
+import './PairedResultGraph.module.scss';
+
+const propTypes = {
+  /**
+   * ID for graph created.
+   */
+  graphID: PropTypes.string.isRequired,
+  /**
+   * Graph configuration object
+   */
+  graphConfig: PropTypes.object.isRequired,
+  /**
+   * data to be displayed in graph
+   */
+  dataset: PropTypes.arrayOf(PropTypes.object),
+  /**
+   * timeout to display multiple data contents in specific time interval.
+   */
+  timeout: PropTypes.arrayOf(PropTypes.number),
+};
+
+const PaiedResultGraph = ({
+  graphConfig, dataset, graphID, timeout,
+}) => {
+  React.useEffect(() => {
+    const graph = Carbon.api.graph(graphConfig);
+    if (dataset) {
+      if (timeout) {
+        dataset.forEach((data, index) => {
+          setTimeout(
+            () => (graph.graphContainer
+              ? graph.loadContent(Carbon.api.pairedResult(data))
+              : ''),
+            timeout[index],
+          );
+        });
+      } else {
+        dataset.forEach((data) => {
+          graph.loadContent(Carbon.api.pairedResult(data));
+        });
+      }
+    }
+  }, [graphConfig, dataset, timeout]);
+
+  return (
+    <div id={`${graphID}-canvasContainer`}>
+      <div id={graphID} />
+    </div>
+  );
+};
+
+PaiedResultGraph.propTypes = propTypes;
+
+export default PaiedResultGraph;
```

Comment:
Have you considered making a base Graph component that accepts the desired Carbon API as a prop? I'm not if future graph types will be the same, but these initial graphs are very similar.

```jsx
const PairedResultGraph = (props) => <Graph {...props} carbonAPI={Carbon.api.pairedResult} />;
const LineGraph = (props) => <Graph {...props} carbonAPI={Carbon.api.line} />;
const BarGraph = (props) => <Graph {...props} carbonAPI={Carbon.api.bar} />;
```

Reply:
We did consider that design choice, but ran into several issues at the time. The current design was what we all agreed upon with Andy. Your solution seems like it may resolve those issues, so can consider this design in the upcoming major version update.